lib/licensed/sources/gradle.rb

```diff
@@ -101,27 +108,50 @@ def enumerate_dependencies
       def gradle_executable
         return @gradle_executable if defined?(@gradle_executable)
         @gradle_executable = begin
-          gradlew = File.join(config.pwd, "gradlew")
-          return gradlew if File.executable?(gradlew)
-          "gradle" if Licensed::Shell.tool_available?("gradle")
-        end
+                               gradlew = File.join(config.pwd, "gradlew")
+                               return gradlew if File.executable?(gradlew)
+                               "gradle" if Licensed::Shell.tool_available?("gradle")
+                             end
       end
 
       # Returns the configurations to include in license generation.
       # Defaults to ["runtime", "runtimeClasspath"]
       def configurations
         @configurations ||= begin
-          if configurations = config.dig("gradle", "configurations")
-            Array(configurations)
-          else
-            DEFAULT_CONFIGURATIONS
-          end
+                              if configurations = config.dig("gradle", "configurations")
+                                Array(configurations)
+                              else
+                                DEFAULT_CONFIGURATIONS
+                              end
+                            end
+      end
+
+      def self.should_add_gradle_license_report_plugins_block?(gradle_build_file)
+        !/dependency-license-report/.match(gradle_build_file)
+      end
+
+      def self.build_gradle_contains_plugins_block?(gradle_build_file)
+        /plugins/.match(gradle_build_file)
+      end
+
+      def self.add_gradle_license_report_plugins_block?(gradle_build_file)
+        if(Licensed::Sources::Gradle.build_gradle_contains_plugins_block?(gradle_build_file))
```

Comment:
can you inline the check that occurs in the called method to make the code a bit simpler?
```suggestion
        if gradle_build_file.include?("plugins")
```